test/edu/ucsb/nceas/metacat/dataone/D1AuthHelperTest.java

```diff
@@ -157,44 +182,293 @@ public void setUp() throws Exception {
         cn1CNSession = new Session();
         cn1CNSession.setSubject(TypeFactory.buildSubject("cn1Subject"));
 
+        // Create a session with a null subject
+        nullSession = new Session();
+        nullSession.setSubject(TypeFactory.buildSubject(null));
+
+        // Create a session without setting the subject
+        missingSubjectSession = new Session();
+
+        // Create a session with an empty subject value
+        emptySubjectSession = new Session();
+        emptySubjectSession.setSubject(TypeFactory.buildSubject(""));
+
+        // Create Metacat Admin session
+        metacatAdminSession = new Session();
+        metacatAdminSession.setSubject(TypeFactory.buildSubject("http://orcid.org/0000-0002-6076-8092"));
+
+        // Create a second Metacat Admin session
+        metacatAdminOtherSession = new Session();
+        metacatAdminOtherSession.setSubject(TypeFactory.buildSubject("http://orcid.org/0000-0003-0077-4738"));
+
+        // Create a Local Node session
+        localNodeSession = new Session();
+        localNodeSession.setSubject(TypeFactory.buildSubject("CN=urn:node:METACAT1,DC=dataone,DC=org"));
+
+        // Create an unauthorized session
+        notAuthorizedSession = new Session();
+        notAuthorizedSession.setSubject(TypeFactory.buildSubject("notAFriend"));
+    }
+
+    /**
+     * Check expandRightsHolder returns true with approved subject, ex. the Session subject is a
+     * member of the rights holder group.
+     */
+    @Test
+    public void testExpandRightsHolder() throws Exception {
+        // expandRightsHolder() is a public static method
+        // D1Client contains methods that make network calls, which is to be mocked
+        try (MockedStatic<D1Client> mockD1Client = Mockito.mockStatic(D1Client.class)) {
+            // Create a member list, to be added to the group
+            List<Subject> hasMemberList = new ArrayList<>();
+            Subject testGroupMember = new Subject();
+            testGroupMember.setValue("testGroupMember");
+            hasMemberList.add(testGroupMember);        // Bogus value
+            hasMemberList.add(sysmeta.getSubmitter()); // The matching member
+            rightsHolderGroup.setHasMemberList(hasMemberList);
+
+            SubjectInfo mockSInfo = Mockito.mock(SubjectInfo.class);
+            List<Group> groups = new ArrayList<>();
+            groups.add(rightsHolderGroup);
+            when(mockSInfo.getGroupList()).thenReturn(groups);
+
+            CNode mockCN = Mockito.mock(CNode.class);
```

Comment:
these lines are repeated verbatim in 2 other tests. Can we extract into a private method?

(bonus points for making it work with the 4th method, too ;-) )

Reply:
Extracted, and refactored the relevant junit tests.